sources/packages/api/src-sql/SINValidations/Alter-columns-for-esdc-sin-validations.sql

```diff
@@ -0,0 +1,128 @@
+-- Remove columns from CRA SIN validation schema.
+ALTER TABLE
+  sims.sin_validations DROP COLUMN IF EXISTS request_status_code,
+  DROP COLUMN IF EXISTS match_status_code,
+  DROP COLUMN IF EXISTS sin_match_status_code,
+  DROP COLUMN IF EXISTS surname_match_status_code,
+  DROP COLUMN IF EXISTS given_name_match_status_code,
+  DROP COLUMN IF EXISTS dob_match_status_code;
+
+-- Adding SIN column, previously it was just on the student table and from now on
+-- it will be the only this the sims.sin_validations.
+ALTER TABLE
+  sims.sin_validations
+ADD
+  -- This will be changed to NOT NULL in the end of the script.
+  COLUMN IF NOT EXISTS sin VARCHAR(9),
+ADD
+  COLUMN IF NOT EXISTS temporary_sin BOOLEAN NOT NULL GENERATED ALWAYS AS (
+    CASE
+      WHEN sin LIKE '9%' THEN TRUE
+      ELSE false
+    END
+  ) STORED;
+
+-- Adding a missing column for history (first name, last name and others are already present).
+ALTER TABLE
+  sims.sin_validations
+ADD
+  COLUMN IF NOT EXISTS gender_sent VARCHAR(10);
+
+-- New columns added based on the new ESDC schema.
+ALTER TABLE
+  sims.sin_validations
+ADD
+  COLUMN IF NOT EXISTS sin_status CHAR(1),
+ADD
+  COLUMN IF NOT EXISTS valid_sin_check CHAR(1),
+ADD
+  COLUMN IF NOT EXISTS valid_birthdate_check CHAR(1),
+ADD
+  COLUMN IF NOT EXISTS valid_first_name_check CHAR(1),
+ADD
+  COLUMN IF NOT EXISTS valid_last_name_check CHAR(1),
+ADD
+  COLUMN IF NOT EXISTS valid_gender_check CHAR(1),
+ADD
+  COLUMN IF NOT EXISTS sin_expire_date DATE;
+
+-- New columns added to allow audit of manual SIN manipulation.
+ALTER TABLE
+  sims.sin_validations
+ADD
+  COLUMN IF NOT EXISTS sin_edited_by INT REFERENCES sims.users (id) ON DELETE
+SET
+  NULL,
+ADD
+  COLUMN IF NOT EXISTS sin_edited_date TIMESTAMP WITH TIME ZONE,
+ADD
+  COLUMN IF NOT EXISTS sin_edited_note_id INT REFERENCES sims.notes (id) ON DELETE
+SET
+  NULL,
+ADD
+  COLUMN IF NOT EXISTS expired_date_edited_by INT REFERENCES sims.users (id) ON DELETE
+SET
+  NULL,
+ADD
+  COLUMN IF NOT EXISTS expired_date_edited_date TIMESTAMP WITH TIME ZONE,
+ADD
+  COLUMN IF NOT EXISTS expired_date_edited_note_id INT REFERENCES sims.notes (id) ON DELETE
+SET
+  NULL;
+
+-- Copy SIN from student to the new sin column on sims.sin_validations.
+UPDATE
```

Comment:
Unless there is a strong reason I would like to avoid adding more files to the migration. Also, the DDL and DML are happening in a specific order to make it work.